src/app/services/qhana-backend.service.ts

```diff
@@ -273,81 +284,91 @@ export class QhanaBackendService {
      * @returns experiment export poll object with result file link if successful
      */
     public exportExperimentPoll(experimentId: number | string, exportId: number | string): Observable<ExperimentExportPollObject> {
-        return this.http.get(`${this.rootUrl}/experiments/${experimentId}/export/${exportId}`, { observe: 'response', responseType: 'arraybuffer' }).pipe(map(resp => {
-            if (resp.headers.get("Content-Type") == "application/json") {
-                return JSON.parse(new TextDecoder().decode(resp.body as ArrayBuffer));
-            } else if (resp.headers.get("Content-Type") == "application/zip") {
-                let result: ExperimentExportPollObject = {
-                    '@self': `${this.rootUrl}/experiments/${experimentId}/export/${exportId}`,
-                    exportId: Number(exportId),
-                    status: 'SUCCESS',
-                    fileLink: `${this.rootUrl}/experiments/${experimentId}/export/${exportId}`
-                };
-                return result;
-            } else {
-                throw new Error("Experiment poll returned wrong file format: " + resp.headers.get("Content-Type"));
-            }
-        }));
+        return this.callWithRootUrl(
+            rootUrl => this.http.get(`${rootUrl}/experiments/${experimentId}/export/${exportId}`, { observe: 'response', responseType: 'arraybuffer' })).pipe(map(resp => {
+                if (resp.headers.get("Content-Type") == "application/json") {
+                    return JSON.parse(new TextDecoder().decode(resp.body as ArrayBuffer));
+                } else if (resp.headers.get("Content-Type") == "application/zip") {
+                    let result: ExperimentExportPollObject = {
+                        '@self': `${this.rootUrl}/experiments/${experimentId}/export/${exportId}`, // TODO: test if this works with this.rootUrl
```

Comment:
format this to better show nesting levels. Also if you move the whole pipeline into the call of `callWithRootUrl`, then you should have access to the passed `rootUrl` without this.